include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -1489,6 +1512,205 @@ struct OrtApi {
+
+  /**
+   * This API fills populates an empty tensor that was created using CreateSparseTensorAsOrtValue API.
+   * The API will allocate required memory and copy the supplied NNZ values and CSR indices into that memory allocation.
+   * Memory allocation is performed using the allocator that was specified with CreateSparseTensorAsOrtValue.
+   * 
+   * \param ort_value - OrtValue to populate with data
+   * \param mem_info - serves to identify the location of the data to be copied. If the allocator specified 
+   *  at the creation time has memory info that is not the same as mem_info argument to this function a X-device copy will be performed.
+   *  String data is assumed to be on CPU and will only be copied into a CPU allocated buffer.
+   * \param values_shape - pointer to values shape array
+   * \param values_shape_len - length of the values_shape
+   * \param values - pointer to an array of values. For strings, pass const char**.
+   * \param inner_indices_num - number of CSR inner indices
+   * \param inner_indices_data - pointer to a location of CSR inner indices
+   * \param outer_indices_num - number of CSR outer indices
+   * \param outer_indices_data - pointer to a location of CSR outer indices
+   */
+  ORT_API2_STATUS(FillSparseTensorCsr, _Inout_ OrtValue* ort_value, _In_ const OrtMemoryInfo* data_mem_info,
+                  _In_ const int64_t* values_shape, size_t values_shape_len, _In_ const void* values,
+                  size_t inner_indices_num, const int64_t* inner_indices_data,
+                  size_t outer_indices_num, const int64_t* outer_indices_data);
+
+  /**
+   * This API fills populates an empty tensor that was created using CreateSparseTensorAsOrtValue API.
+   * The API will allocate required memory and copy the supplied NNZ values and BlockSparse indices into that memory allocation.
+   * Memory allocation is performed using the allocator that was specified with CreateSparseTensorAsOrtValue.
+   * 
+   * \param ort_value - OrtValue to populate with data
+   * \param mem_info - serves to identify the location of the data to be copied. If the allocator specified 
+   *  at the creation time has memory info that is not the same as mem_info argument to this function a X-device copy will be performed.
+   *  String data is assumed to be on CPU and will only be copied into a CPU allocated buffer.
+   * \param values - structure with values information
+   * \param indices_shape_len - length of the block sparse indices shape
+   * \param indices_shape - pointer to a location of indices shape
+   * \param indices_data - pointer to a location of indices data. Shape will determine the length of the indices data.
+   */
+  ORT_API2_STATUS(FillSparseTensorBlockSparse, _Inout_ OrtValue* ort_value, _In_ const OrtMemoryInfo* data_mem_info,
+                  _In_ const int64_t* values_shape, size_t values_shape_len, _In_ const void* values,
+                  size_t indices_shape_len, const int64_t* indices_shape_data,
```

Comment:
Would it be better for our users if we remain consistent with order of ptr and len?

Reply:
Thx for addressing this.